### PR TITLE
fix(item-sliding): RTL fix for item sliding

### PR DESCRIPTION
**Pull request by sijav**
#### Short description of what this resolves:
this will resolve a but that options would disappear on RTL.

#### Changes proposed in this pull request:

- the class will be placed based on site direction

**Ionic Version**: 3.x

**Fixes**: #11211

### CODE REVIEW

**Comment by sijav** [comment]
This will complete with #11822

**Comment by AmitMY** [comment]
I am not sure about doing it from TS.
In theory, we can fix it from scss [here](https://github.com/driftyco/ionic/blob/master/src/components/item/item-sliding.scss#L107) by manipulating the order like
```
@include ltr() { order: 1 }
@include rtl() { order: -1 }
```

Because:
- It allows us to change the naming `swipe-right` to be correct - `swipe-end`
- It allows real-time direction change without extra effort (if you initiate the view and then change direction your solution would stay in the first direction)

What do you think?

**Comment by sijav** [comment]
@AmitMY Yeah well ... it was just a quick fix for this...

**Comment by AmitMY** [comment]
@sijav Haha not bashing or anything, you are doing great work :) 
This would work, yes, but not for real time, and it will be simple to change so I noted it. I just know that in the future it will annoy me such things don't work in real-time (and some things are just a lot of work so we drop them for now)

This can be merged and fixed in the future, but we all know what happens when we say that :P

**Comment by sijav** [comment]
I know I meant that I didn't think of anything but to announce this bug.

**Comment by sijav** [comment]
@AmitMY I've changed the fix, I'm using scss changes instead of ts changes... let me know what you think.

**Comment by AmitMY** [comment]
@sijav Did you test this? it seems very odd to me. Also that you changed the width, and classes that are not related as I *think* that what needs to be changed is the order, not these classes (which are clones - see https://github.com/ionic-team/ionic/pull/11829)

I believe this selector - "ion-item-sliding.active-swipe-right button[expandable] {" needs a change, like:
```
@include ltr() { order: 1 }
@include rtl() { order: -1 }
```
but don't worry about it, as the `ltr` mixin is not yet available on master..

I think this one should be paused for the moment, and when this - https://github.com/ionic-team/ionic/pull/11635 - gets merged, we should continue discussion

**Comment by sijav** [comment]
@AmitMY Yes I've tested it and it worked great.
And no there's no order matters it just a matter of left and right class that visible the right property as they being slide...

**Comment by sijav** [comment]
@AmitMY this is still buggy but that doesn't have to do with just a simple order the left and right should be for left and right only regardless of being ltr or rtl, I'm working on it and let you know ...
**Conclusion**: This fix is still on development...

**Comment by sijav** [comment]
@AmitMY Could you please review this? this is the only way I could make it work!
Be noticed that now `Right` will place the sliding options to right side of the item and so does `Left` will place the sliding option button to the left side of the item and unless we use start or end this is the only correct way to do it.

File: src/components/item/item-sliding.scss [hunk 1]
```diff
@@ -29,14 +28,32 @@ ion-item-options {
...
   font-size: 14px;
-
   visibility: hidden;
+
+  right: 0;
+
+  @include rtl() {
+    justify-content: flex-start;
+
+    &:not([side=right]) {
```
Comment (sijav) [inline src/components/item/item-sliding.scss]: these `:not([side=right])` are here for default side (when side is not defined in sliding option item) in RTL mode.
In LTR mode there should be positioned on right by default but on RTL they need to be placed left...
In another word when side is not defined it should be placed on `end`

**Comment by AmitMY** [comment]
Can you please redo this with `multi-dir`, `rtl`, and `ltr`? too much has changed :)

**Comment by sijav** [comment]
@AmitMY done, can you review this?

File: src/components/item/item-sliding.scss [hunk 2]
```diff
@@ -17,26 +17,52 @@ ion-item-sliding .item {
 }
 
 ion-item-options {
-  @include position(0, 0, null, null);
-
+  // scss-lint:disable PropertySpelling
```
Comment (AmitMY) [inline src/components/item/item-sliding.scss]: Can you move this comment to inside the mixin?

Comment (sijav) [inline src/components/item/item-sliding.scss]: @AmitMY so I should put it at two places? multi-dir and rtl mixin at the same time?

Comment (AmitMY) [inline src/components/item/item-sliding.scss]: I think it would be better, yes. Just so it is more specific.

**Comment by brandyscarney** [comment]
Thank you! 👍

**Pull request merged**